lib/models/body_weight/weight_entry.dart

```diff
@@ -43,7 +43,7 @@ class WeightEntry {
   WeightEntry copyWith({int? id, int? weight, DateTime? date}) => WeightEntry(
         id: id,
         weight: weight ?? this.weight,
-        date: date ?? this.date,
+        date: date ?? DateTime.now(),
```

Comment:
when you call copyWith(), you expect that it copies the model, and overrides those fields that are given.
therefore this change doesn't make sense. Instead, the callers should just provide whichever date they want to use (DateTime.now())